internal/config/config.go

```diff
@@ -873,22 +873,35 @@ func checkConfig(config *Config) error {
 	}
 
 	// Set defaults
-	for _, registryAuth := range config.DockerRegistriesAuth {
+	for rai, registryAuth := range config.DockerRegistriesAuth {
+		if registryAuth == nil {
+			return errors.Errorf("docker registries auth at index %s is empty", rai)
+		}
+
 		if registryAuth.Type == "" {
 			registryAuth.Type = DockerRegistryAuthTypeBasic
 		}
 	}
 
 	for _, run := range config.Runs {
 		// set auth type to basic if not specified
-		for _, registryAuth := range run.DockerRegistriesAuth {
+		for rai, registryAuth := range run.DockerRegistriesAuth {
+			if registryAuth == nil {
+				return errors.Errorf("run %q, docker registries auth at index %s is empty", run.Name, rai)
+			}
+
 			if registryAuth.Type == "" {
 				registryAuth.Type = DockerRegistryAuthTypeBasic
 			}
 		}
 		for _, task := range run.Tasks {
 			// set auth type to basic if not specified
-			for _, registryAuth := range task.DockerRegistriesAuth {
+			for rai, registryAuth := range task.DockerRegistriesAuth {
+				if registryAuth == nil {
+					return errors.Errorf("run %q, task %q, docker registries auth at index %s is empty", run.Name, task.Name, rai)
```

Comment:
remove `index` since it's a string and not a number: 

`return errors.Errorf("run %q, task %q, docker registries auth %q is empty" , run.Name, task.Name, raName)`